sdk/compute/Microsoft.Azure.Management.Compute/tests/ScenarioTests/VMTestBase.cs

```diff
@@ -888,6 +888,41 @@ protected string CreateProximityPlacementGroup(string rgName, string ppgName)
             return vm;
         }
 
+        protected DedicatedHostGroup CreateDedicatedHostGroup( string rgName, string dedicatedHostGroupName)
+        {
+            m_ResourcesClient.ResourceGroups.CreateOrUpdate(
+                   rgName,
+                   new ResourceGroup
+                   {
+                       Location = m_location,
+                       Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } }
+                   });
+
+            DedicatedHostGroup dedicatedHostGroup = new DedicatedHostGroup()
+            {
+                Location = m_location,
+                Zones = new List<string> { "1" }
+            };
+            return m_CrpClient.DedicatedHostGroups.CreateOrUpdate(rgName, dedicatedHostGroupName, dedicatedHostGroup);
+        }
+
+        protected DedicatedHost CreateDedicatedHost(string rgName, string dedicatedHostGroupName, string dedicatedHostName)
+        {
+            //Check if DedicatedHostGroup already exist and if does not exist, create one.
+            DedicatedHostGroup existingDHG = m_CrpClient.DedicatedHostGroups.Get(rgName, dedicatedHostGroupName);
+            if (existingDHG == null)
+            {
+                existingDHG = CreateDedicatedHostGroup(rgName, dedicatedHostGroupName);
+            }
+            return m_CrpClient.DedicatedHosts.CreateOrUpdate(rgName, dedicatedHostGroupName, dedicatedHostName,
+                new DedicatedHost()
+                {
+                    Location = m_location,
+                    Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } },
```

Comment:
Likewise you need to provide platformFaultDomain if platformFaultDomainCount > 1.

Reply:
It is optional in Swagger spec.